osmomath/decimal.go

```diff
@@ -972,3 +972,23 @@ func (x BigDec) CustomBaseLog(base BigDec) BigDec {
 
 	return y
 }
+
+// PowerInteger takes a given decimal to an integer power
+// and returns the result. Non-mutative. Uses square and multiply
+// algorithm for performing the calculation.
+func (d BigDec) PowerInteger(power uint64) BigDec {
```

Comment:
Do we have mutative multiplication on bigdec?

If so, lets make a PowerIntegerMut, have this create a copy, and then run PowerIntegerMut on the copy

Reply:
There is no mutative multiplication. This would require introducing `MulMut` and refactoring `Mul` to utilize that.

I'm going to implement that in a separate PR. As soon as the new PR is merged, I will rebase this and implement the requested change.

Reply:
https://github.com/osmosis-labs/osmosis/pull/3678